### ISSUE TITLE
rational.jl: rationalize NaN

### ISSUE THREAD

**Issue by ederag** [JuliaLang/julia#26823]
Reading the documentation about [conversion and promotion](https://docs.julialang.org/en/stable/manual/conversion-and-promotion/#man-rational-conversion-1) of 0.6.2, the `isnan` line looked weird:

```
function convert(::Type{Rational{T}}, x::AbstractFloat, tol::Real) where T<:Integer
    if isnan(x); return zero(T)//zero(T); end
```

The issue is multiple:
- in the actual source `convert` has been changed to `rationalize`. The doc could be updated, for clarity.
- Trying that in julia 0.6.2 yields an error message that could be puzzling before looking at the source.
```
julia> rationalize(Int64, NaN, 1e-16)
ERROR: ArgumentError: invalid rational: zero(Int64)//zero(Int64)
Stacktrace:
 [1] Rational{Int64}(::Int64, ::Int64) at ./rational.jl:13
 [2] rationalize(::Type{Int64}, ::Float64, ::Float64) at ./rational.jl:131
```
What about replacing ``zero(T)//zero(T)`` with something like ``throw(ArgumentError("received NaN, not allowed for integer-related types")`` ?

Here is the corresponding line in the repo:
https://github.com/JuliaLang/julia/blob/c0e6b5ba8125f86d377edc2c8297365bf6427caf/base/rational.jl#L130

This is not https://github.com/JuliaLang/julia/pull/25702.

**Comment by ederag** [comment]
@JeffBezanson If the doc label means that the problem is only in the doc, let me emphasize that the code itself seems wrong, in my julia newbie opinion. Getting ``ERROR: ArgumentError: invalid rational: zero(Int64)//zero(Int64)`` for a NaN argument is surprising.

**Comment by JeffBezanson** [comment]
As it turns out the manual no longer contains this particular code, so the doc part of the issue is mostly moot.

It's ambiguous whether the doc label means "doc only", but people often tag issues with all relevant labels, so I'd say it's closer to just meaning "doc changes needed here".

**Comment by ederag** [comment]
@JeffBezanson thanks for the explanation. @stevengj proposed a nice improvement, in PR #26875.

**Comment by stevengj** [comment]
Closed  by #26875.
